### ISSUE TITLE
`fromGethGenesis` sets `DEFAULT_HARDFORK` to `merge` when no `hardfork` is provided.

### ISSUE THREAD

**Issue by tenthirtyone** [ethereumjs/ethereumjs-monorepo#2414]
There may be a small bug in the way `fromGethGenesis` sets the default hardfork when creating a new `Common`. 

The `parseGethGenesis` function it relies on to convert from `genesis.json` into `CommonOpts` relies on a [forkmap](https://github.com/ethereumjs/ethereumjs-monorepo/blob/5738e2762b260df279a26d924d12348ce8a9d3ce/packages/common/src/utils.ts#L83) to determine which hardforks are available to the chain.

Providing `fromGethGenesis` with a hardfork that is not defined in `genesis.json` will, correctly, throw an error. E.g. specifying `london` with a matching `londonBlock` in `genesis.json` throws. However, if a `hardfork` is not provided Common's [constructor](https://github.com/ethereumjs/ethereumjs-monorepo/blob/9d470d336fa2cd606dbac31c4c6f8a6a1981f878/packages/common/src/common.ts#L220) sets `DEFAULT_HARDFORK` to `merge`, regardless of whether a corresponding `mergeForkBlock` was parsed from `genesis.json`

I believe the default behavior to choose the most recent fork makes sense. Since the available hardforks are defined by `genesis.json`  when using `fromGethGenesis` I think it might be better for the default hardfork value to be set to the most recent hardfork defined in `genesis.json` by block height.  

To reproduce add these tests to https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/common/test/utils.spec.ts 

``` javascript
  t.test('Throws', (t) => {
    const json = require(`../../client/test/testdata/geth-genesis/testnet.json`)

    // This is allowed - but no mergeForkBlock exists in testnet.json
    const common1 = Common.fromGethGenesis(json, {
      chain: 'customChain1',
    })

    //common1.hardfork() === 'merge'

    // This will throw - Hardfork with name merge not supported
    const common2 = Common.fromGethGenesis(json, {
      chain: 'customChain2',
      hardfork: 'merge',
    })


  })

  t.test('Not equal', (t) => {
    const json = require(`../../client/test/testdata/geth-genesis/testnet.json`)

    const common1 = Common.fromGethGenesis(json, {
      chain: 'customChain1',
    })

    const common2 = Common.fromGethGenesis(json, {
      chain: 'customChain2',
      hardfork: 'london',
    })

    // common1 === merge
    // common2 === london
    t.equals(common1.hardfork(), common2.hardfork())
  })
```

**Comment by holgerd77** [comment]
Hi there, thanks for reporting, also just read the comment in the chat. To add here: generally we very much rely on the assumption within the EthereumJS libraries that a not-further-hardfork-referencing Common initializes with the default HF (currently: Merge), regardless what initialization method for Common has been used.

So always setting to the most recent fork is deviating too much from this behavior and can (likely will) have unwanted side effects.

So I would adopt here a bit to say that in a first iteration the default HF should tried to be set, and if this fails we can take the most recent fork instead.

**Comment by tenthirtyone** [comment]
@holgerd77 That makes sense, thanks for sharing your thoughts.

**Comment by gabrocheleau** [comment]
I believe this has been addressed by #2470
